Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,196 @@
+                                 title: LocalizedStrings.receivingAccount,
+                                 selectedWallet: selectedWallet,
+                                 selectedAccount: self.selectedAccount,
+                                 callback: self.updateSelectedAccount)
+    }
+
+    private func checkSyncStatus() {
+        self.moreMenuButton.isEnabled = false
+    
+        let accountsFilterFn: (DcrlibwalletAccount) -> Bool = { $0.totalBalance > 0 || $0.name != "imported" }
+        guard let wallet = WalletLoader.shared.wallets.map({ Wallet.init($0, accountsFilterFn: accountsFilterFn) }).first,
+            let account = wallet.accounts.first else {
+                self.mainContentViewHolder.isHidden = true
+                self.syncInProgressLabel.isHidden = false
+                return
+        }
+
+        self.updateSelectedAccount(wallet.id, account)
+
+        mainContentViewHolder.isHidden = false
+        syncInProgressLabel.isHidden = true
+        self.moreMenuButton.isEnabled = true
+    }
+
+    private func generateNewAddress() {
+        self.oldAddress = self.walletAddressLabel.text!
+        self.getNextAddress()
+    }
+
+    private func updateWalletAddressAndQRCode(receiveAddress: String) {
+        DispatchQueue.main.async {
+            self.walletAddressLabel.text = receiveAddress
+            self.addressQRCodeImageView.image = self.generateQRCodeFor(
+                with: receiveAddress,
+                forImageViewFrame: self.addressQRCodeImageView.frame
+            )
+        }
+    }
+
+    @objc private func getNextAddress() {
+        if let wallet = self.selectedWallet,
+            let account = self.selectedAccount {
+                let receiveAddress = wallet.nextAddress(account.number, error: nil)
+                if self.oldAddress != receiveAddress {
+                    self.updateWalletAddressAndQRCode(receiveAddress: receiveAddress)
+                } else if receiveAddress != "" {
+                    self.getNextAddress()
+                }
+        }
+    }
+
+    private func generateQRCodeFor(with addres: String, forImageViewFrame: CGRect) -> UIImage? {
+        guard let addrData = addres.data(using: String.Encoding.utf8) else {
+            return nil
+        }
+
+        // Color code and background
+        guard let colorFilter = CIFilter(name: "CIFalseColor") else { return nil }
+        let filter = CIFilter(name: "CIQRCodeGenerator")
+        filter?.setValue(addrData, forKey: "inputMessage")
+
+        let foregroundColor = CIColor.black
+        let backgroundColor = CIColor.clear
+
+        colorFilter.setDefaults()
+        colorFilter.setValue(filter!.outputImage, forKey: "inputImage")
+        colorFilter.setValue(foregroundColor, forKey: "inputColor0")
+        colorFilter.setValue(backgroundColor, forKey: "inputColor1")
+
+        if let imgQR = colorFilter.outputImage {
+            var tempFrame: CGRect? = forImageViewFrame
+            if tempFrame == nil {
```

Comment:
`forImageViewFrame` which should now be `self.addressQRCodeImageView.frame` is never nil, this if block can be deleted.

This `tempFrame` variable is actually useless. Please delete.